polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -2396,6 +2396,287 @@ ValueCategory MLIRScanner::EmitPromotedScalarExpr(Expr *E,
   return Visit(E);
 }
 
+ValueCategory MLIRScanner::EmitScalarCast(ValueCategory Src, QualType SrcType,
+                                          QualType DstType, mlir::Type SrcTy,
+                                          mlir::Type DstTy) {
+  assert(!SrcType->isAnyComplexType() && !DstType->isAnyComplexType() &&
```

Comment:
Should we early exit for `SrcTy == DstTy`?

Reply:
That should not happen in fact. I'll add an assert instead.